### PR TITLE
[FLINK-11560]Translate "Flink Applications" page into Chinese

### PR DESCRIPTION
**Pull request by bahuasheng**
Translate flink-application.zh.md

### CODE REVIEW

**Comment by wuchong** [comment]
Hi @bahuasheng , please correct the title to corresponding JIRA title and follow the [commit message guideline](https://flink.apache.org/contribute-code.html#coding-guidelines).

**Comment by bahuasheng** [comment]
Hi, Jark Wu,
Thanks for your reminding. I've corrected the title according to JIRA title.

best,
Stephen

**Comment by XuQianJin-Stars** [comment]
hi @bahuasheng  Thanks for the effort.

Some formatting issues can refer to the Flink Translation Specifications:
https://docs.google.com/document/d/1zhGyPU4bVJ7mCIdUTsQr9lHLn4vltejhngG_SOV6g1w/edit#heading=h.by2zo7lwgzr5

Such as:
1. Space between Chinese characters, English characters and Numbers
2. Use "you" instead of "you"(一般用“你”，而不是“您”)
3. Free translation is better than literal translation

File: flink-applications.zh.md [hunk 1]
```diff
@@ -6,197 +6,209 @@ title: "Apache Flink 是什么？"
...
       <span class="glyphicon glyphicon-chevron-right"></span> &nbsp;
-      <a href="{{ site.baseurl }}/flink-operations.html">Operations</a>
+      <a href="{{ site.baseurl }}/zh/flink-operations.html">操作</a>
     </h2>
   </div>
 </div>
 <hr/>
 
-Apache Flink is a framework for stateful computations over unbounded and bounded data streams. Flink provides multiple APIs at different levels of abstraction and offers dedicated libraries for common use cases.
+Apache Flink是一个用于对无界和有界数据流进行有状态计算的框架。 Flink在不同的抽象级别提供多种API，并为常见用例提供专用库。
```
Comment (XuQianJin-Stars) [inline flink-applications.zh.md]: Apache Flink是一个用 -> Apache Flink 是一个用

File: flink-applications.zh.md [hunk 2]
```diff
@@ -6,197 +6,209 @@ title: "Apache Flink 是什么？"
...
 
-Here, we present Flink's easy-to-use and expressive APIs and libraries.
+本小节，我们将介绍Flink框架中易于使用和富有表现力的API和函数库。
```
Comment (XuQianJin-Stars) [inline flink-applications.zh.md]: 我们将介绍Flink框架中易于使用和富有表现力的API -> 我们将介绍 Flink 框架中易于使用和富有表现力的API 和

Spaces should be added between all English and Chinese below.

File: flink-applications.zh.md [hunk 3]
```diff
@@ -6,197 +6,209 @@ title: "Apache Flink 是什么？"
...
 
-* **Multiple State Primitives**: Flink provides state primitives for different data structures, such as atomic values, lists, or maps. Developers can choose the state primitive that is most efficient based on the access pattern of the function.
-* **Pluggable State Backends**: Application state is managed in and checkpointed by a pluggable state backend. Flink features different state backends that store state in memory or in [RocksDB](https://rocksdb.org/), an efficient embedded on-disk data store. Custom state backends can be plugged in as well.
-* **Exactly-once state consistency**: Flink's checkpointing and recovery algorithms guarantee the consistency of application state in case of a failure. Hence, failures are transparently handled and do not affect the correctness of an application.
-* **Very Large State**: Flink is able to maintain application state of several terabytes in size due to its asynchronous and incremental checkpoint algorithm.
-* **Scalable Applications**: Flink supports scaling of stateful applications by redistributing the state to more or fewer workers.
+* **多状态原语**: Flink 为不同的数据结构提供状态原语，例如原子值，列表或映射。 开发人员可以根据函数的访问模式来选择最有效的状态原语。
```
Comment (wuchong) [inline flink-applications.zh.md]:
```suggestion
* **多种状态基础类型**: Flink 为不同的数据结构提供了状态基础类型，例如原子值、List 列表、Map 映射。开发人员可以根据函数的访问模式来选择最有效的状态基础类型。
```

File: flink-applications.zh.md [hunk 4]
```diff
@@ -6,197 +6,209 @@ title: "Apache Flink 是什么？"
...
+* **可插拔的状态后端**: 应用程序状态由可插拔的状态后端进行管理和检查。 Flink 具有不同的状态后端，可以将状态存在内存或[RocksDB][1]，后者是一种高效的嵌入式磁盘数据存储引擎。 另外，自定义状态后端可以嵌入 Flink 中。
```
Comment (wuchong) [inline flink-applications.zh.md]:
```suggestion
* **插件化的状态后端**: 应用程序的状态由插件化的状态后端进行管理并执行检查点。Flink 提供了不同的状态后端，可以将状态存在内存或[RocksDB](https://rocksdb.org/)，后者是一种高效的嵌入式磁盘数据存储引擎。另外，也可以开发自定义的状态后端插件。
```

File: flink-applications.zh.md [hunk 5]
```diff
@@ -6,197 +6,209 @@ title: "Apache Flink 是什么？"
...
+* **Exactly-once状态一致性**: Flink 的检查点和恢复算法可以保证在发生故障时应用程序状态的一致性。 因此，故障是透明处理的，不会影响应用程序的正确性。
+* **大规模状态**: 由于其异步和增量检查点算法，Flink能够维护几兆字节的应用程序状态
```
Comment (wuchong) [inline flink-applications.zh.md]:
```suggestion
* **超大状态**: 基于 Flink 的异步和增量检查点算法，Flink 能够维护几 TB 的应用程序状态。
```
terabytes 不是 "兆字节"

File: flink-applications.zh.md [hunk 6]
```diff
@@ -6,197 +6,209 @@ title: "Apache Flink 是什么？"
...
+* **可扩展应用程序**: Flink 通过将状态重新分配给更多或更少的计算节点实现有状态应用程序的扩展。
```
Comment (wuchong) [inline flink-applications.zh.md]:
```suggestion
* **可伸缩的应用程序**: Flink 通过将状态重新分配给更多或更少的计算节点来实现有状态应用程序的可伸缩性。
```

File: flink-applications.zh.md [hunk 7]
```diff
@@ -6,197 +6,209 @@ title: "Apache Flink 是什么？"
...
 
-### Time
+### 时间
 
-Time is another important ingredient of streaming applications. Most event streams have inherent time semantics because each event is produced at a specific point in time. Moreover, many common stream computations are based on time, such as windows aggregations, sessionization, pattern detection, and time-based joins. An important aspect of stream processing is how an application measures time, i.e., the difference of event-time and processing-time.
+时间是流应用程序的另一个重要组成部分。大多数事件流都具有固有的时间语义，因为每个事件都是在特定时间点生成的。 此外，许多常见的流计算（例如时间窗口内聚合, sessionization，模式检测和基于时间的连接等）是基于时间的。 流处理的一个重要方面是应用程序如何处理时间，即如何处理事件时间和处理时间的差异。
```
Comment (wuchong) [inline flink-applications.zh.md]:
```suggestion
时间是流应用程序的另一个重要组成部分。大多数事件流都具有固有的时间语义，因为每个事件都是在特定时间点生成的。 此外，许多常见的流计算（例如窗口聚合、会话计算、模式检测和基于时间的Join等）是基于时间的。 流处理的一个重要方面是应用程序如何处理时间，即事件时间和处理时间的区别。
```

File: flink-applications.zh.md [hunk 8]
```diff
@@ -6,197 +6,209 @@ title: "Apache Flink 是什么？"
...
-* **Watermark Support**: Flink employs watermarks to reason about time in event-time applications. Watermarks are also a flexible mechanism to trade-off the latency and completeness of results.
-* **Late Data Handling**: When processing streams in event-time mode with watermarks, it can happen that a computation has been completed before all associated events have arrived. Such events are called late events. Flink features multiple options to handle late events, such as rerouting them via side outputs and updating previously completed results.
-* **Processing-time Mode**: In addition to its event-time mode, Flink also supports processing-time semantics which performs computations as triggered by the wall-clock time of the processing machine. The processing-time mode can be suitable for certain applications with strict low-latency requirements that can tolerate approximate results.
+* **事件时间模式**: 应用程序基于事件的时间戳处理具有事件时间语义的流数据。 因此，无论是处理录制的事件还是实时的事件，通过事件时间模式处理都能够得到准确和一致的结果
+* **水印支持**: Flink 采用水印来推断事件时间应用中的时间。 水印同时也是一种灵活的机制用以权衡结果的延迟和完整性。
```
Comment (wuchong) [inline flink-applications.zh.md]: "水印" is not a good translation. We prefer not to translate it.

File: flink-applications.zh.md [hunk 9]
```diff
@@ -6,197 +6,209 @@ title: "Apache Flink 是什么？"
...
-* **Watermark Support**: Flink employs watermarks to reason about time in event-time applications. Watermarks are also a flexible mechanism to trade-off the latency and completeness of results.
-* **Late Data Handling**: When processing streams in event-time mode with watermarks, it can happen that a computation has been completed before all associated events have arrived. Such events are called late events. Flink features multiple options to handle late events, such as rerouting them via side outputs and updating previously completed results.
-* **Processing-time Mode**: In addition to its event-time mode, Flink also supports processing-time semantics which performs computations as triggered by the wall-clock time of the processing machine. The processing-time mode can be suitable for certain applications with strict low-latency requirements that can tolerate approximate results.
+* **事件时间模式**: 应用程序基于事件的时间戳处理具有事件时间语义的流数据。 因此，无论是处理录制的事件还是实时的事件，通过事件时间模式处理都能够得到准确和一致的结果
```
Comment (wuchong) [inline flink-applications.zh.md]:
```suggestion
* **事件时间模式（Event-time）**: 应用程序基于事件的时间戳处理具有事件时间语义的流数据。因此，无论是处理历史记录的事件还是实时的事件，通过事件时间模式处理都能够得到准确和一致的结果
```

File: flink-applications.zh.md [hunk 10]
```diff
@@ -6,197 +6,209 @@ title: "Apache Flink 是什么？"
...
+* **水印支持**: Flink 采用水印来推断事件时间应用中的时间。 水印同时也是一种灵活的机制用以权衡结果的延迟和完整性。
+* **延迟数据处理**:当使用水印在事件时间模式下处理流时，可能会发生在所有相关事件到达之前就完成计算。 这类事件被称为延迟事件。 Flink 有多种方式来处理延迟事件，例如通过额外输出实现重新路由这些事件，以及更新之前完成的结果
```
Comment (wuchong) [inline flink-applications.zh.md]:
```suggestion
* **迟到数据处理**: 当基于事件时间和 watermark 处理数据流时，可能会发生在所有相关事件到达之前就完成计算。这类事件被称为迟到事件。Flink 有多种方式来处理迟到事件，例如通过 side output 重新路由这些事件或者更新之前完成的结果。
```

我觉得 "延迟" 在这里可能会引起读者的歧义，是代表动词呢，还是形容词呢？ 所以我建议使用 "迟到"，会更清晰些。

File: flink-applications.zh.md [hunk 11]
```diff
@@ -6,197 +6,209 @@ title: "Apache Flink 是什么？"
...
+* **处理事件模型**:除了事件时间模式之外，Flink 还支持处理时间语义，该处理时间语义的执行由处理机器的时钟触发。 处理时间模式适用于具有严格的低延迟要求的某些应用，这些要求可以容忍近似的结果。
```
Comment (wuchong) [inline flink-applications.zh.md]:
```suggestion
* **处理时间模型（Processing-time）**: 除了事件时间模式之外，Flink 还支持处理时间语义，该处理时间语义的执行由处理机器的时钟触发。处理时间模式适用于具有严格的低延迟要求的某些应用，这些要求可以容忍近似的结果。
```

专业术语我们建议在第一次出现时，加上英文原文。

File: flink-applications.zh.md [hunk 12]
```diff
@@ -6,197 +6,209 @@ title: "Apache Flink 是什么？"
...
 
-## Layered APIs
+## 分层的API
```
Comment (wuchong) [inline flink-applications.zh.md]:
```suggestion
## 分层 API
```

分层 API ?

**Review by wuchong — COMMENTED** [review]
Hi @bahuasheng , thanks for your contribution. I left some suggestions in Chinese.

我并没有完全review完整个PR，目前发现有以下几个方面的问题。

1. 不要在句号后加空格（即使后面跟的是一个英文单词）。
2. 请follow markdown 的链接语法。不要将链接置于文末。
3. 请不要修改原有代码的格式。
4. 建议在本地预览下翻译后的渲染样式和格式与原文是否一直。本地预览方式见： https://flink.apache.org/zh/improve-website.html

**Comment by wuchong** [comment]
We have another pull request for this which is already close to merge.  see #215

File: flink-applications.zh.md [hunk 13]
```diff
@@ -6,197 +6,209 @@ title: "Apache Flink 是什么？"
 <div class="row">
   <div class="col-sm-12" style="background-color: #f8f8f8;">
     <h2>
-      <a href="{{ site.baseurl }}/flink-architecture.html">Architecture</a> &nbsp;
```
Comment (Alsultany84) [inline flink-applications.zh.md:9]: https://cwiki.apache.org/confluence/display/FLINK/Flink+Translation+Specifications

**Pull request closed without merging**